src/pairinteraction_gui/worker.py

```diff
@@ -104,16 +106,46 @@ def __init__(self, fn_name: str, *args: Any, **kwargs: Any) -> None:
     @property
     def pool(self) -> Pool:
         if self._pool is None:
-            return self.create_pool()
-        return self._pool
+            self.create_pool()
+        return self._pool  # type: ignore [return-value]
+
+    @classmethod
+    def create_pool(cls, n_processes: int = 1, *, run_async: bool = False) -> None:
+        """Create a pool of processes."""
+        if isinstance(cls._pool, Pool):
+            if cls._pool._processes == n_processes:  # type: ignore [attr-defined]
+                return
+            raise ValueError("Pool already created with different number of processes.")
+        if cls._async_worker is not None:
+            if not run_async:
+                logger.debug("Waiting for async worker to finish.")
+                cls._async_worker.join()
+                logger.debug("Async worker finished.")
+            return
+
+        if not run_async:
```

Comment:
Is it beneficial not to use async in some scenarios? If this is not the case and we only use async, I would remove the non-async option.